src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/DSL.kt

```diff
@@ -184,6 +147,18 @@ interface BookingDsl {
     auditModuleName: String? = null,
     iepDateTime: LocalDateTime = LocalDateTime.now(),
   )
+
+  @CourseAllocationDslMarker
+  fun courseAllocation(
+    courseActivity: CourseActivity,
+    startDate: String? = "2022-10-31",
+    programStatusCode: String = "ALLOC",
+    endDate: String? = null,
+    payBands: MutableList<OffenderProgramProfilePayBandBuilder> = mutableListOf(),
+    endReasonCode: String? = null,
+    endComment: String? = null,
+    dsl: CourseAllocationDsl.() -> Unit = {},
```

Comment:
Interesting that as the course allocation has 2 parents it needs 2 different abstract methods in 2 interfaces. Each of thse methods currently has the same defaults but they could be different - not sure if this is an issue.